webui/react/src/components/SelectFilter.tsx

```diff
@@ -0,0 +1,63 @@
+import { Select } from 'antd';
+import { SelectValue } from 'antd/es/select';
+import React, { PropsWithChildren, useCallback } from 'react';
+
+import Icon from './Icon';
+import css from './SelectFilter.module.scss';
+
+interface Props {
+  enableSearchFilter?: boolean;
+  label: string;
+  onSelect?: (value: SelectValue) => void;
+  placeholder?: string | React.ReactNode;
+  showSearch?: boolean;
+  style?: React.CSSProperties;
+  value?: SelectValue;
+}
+
+const defaultProps = {
+  enableSearchFilter: true,
+  showSearch: true,
+};
+
+export const ALL_VALUE = 'all';
+
+const SelectFilter: React.FC<PropsWithChildren<Props>> = ({
+  children, enableSearchFilter, label, onSelect, placeholder, showSearch, style, value,
```

Comment:
nit: I personally lean toward not deconstructing especially when there are more than 2 or 3 props. I think I prefer using props but maybe use `p` to make it shorter? that would quickly tell us where each variable is coming from which is important for rerendering and wouldn't pollute the component scope

Reply:
That's a good idea, I will deconstruct only the ones that are used as direct hook dependencies such as `useCallback`, `useMemo`, and `useEffect`.

Reply:
ended up using just plain ol' `props` since none of them were dependencies +1